bitnami/elasticsearch/templates/data/statefulset.yaml

```
@@ -85,7 +85,7 @@ spec:
       {{- end }}
       {{- if or (or .Values.initContainers .Values.data.initContainers) .Values.security.enabled .Values.sysctlImage.enabled (and .Values.volumePermissions.enabled .Values.data.persistence.enabled) }}
```

If I am not wrong, we should set the initContainers if users set any of these values:
* `initContainers`
* `data.initContainers`
* `enableDefaultInitContainers` and `security.enabled ` 
* `enableDefaultInitContainers` and `sysctlImage.enabled`
* `volumePermissions.enabled` and `.data.persistence.enabled`

So the condition should be:

```suggestion
      {{- if or .Values.initContainers .Values.data.initContainers (and  .Values.enableDefaultInitContainers (or .Values.security.enabled .Values.sysctlImage.enabled)) (and .Values.volumePermissions.enabled .Values.data.persistence.enabled) }}
```